File: app/src/main/java/com/ozhan/mustafa/howl/core/registration/RegisterContract.java
package com.ozhan.mustafa.howl.core.registration;

import android.app.Activity;

import com.google.firebase.auth.FirebaseUser;

/**
 * Created by Mustafa Özhan on 3/4/17 at 12:57 PM.
 */

public interface RegisterContract {
    interface View {
        void onRegistrationSuccess(FirebaseUser firebaseUser);

        void onRegistrationFailure(String message);
    }

    interface Presenter {
        void register(Activity activity, String email, String password);
    }

    interface Interactor {
        void performFirebaseRegistration(Activity activity, String email, String password);
    }

    interface OnRegistrationListener {
        void onSuccess(FirebaseUser firebaseUser);

        void onFailure(String message);
    }
}

File: app/src/main/java/com/ozhan/mustafa/howl/core/registration/RegisterContract.kt
package com.ozhan.mustafa.howl.core.registration

import android.app.Activity

import com.google.firebase.auth.FirebaseUser

/**
 * Created by Mustafa Özhan on 3/4/17 at 12:57 PM.
 */

interface RegisterContract {
    interface View {
        fun onRegistrationSuccess(firebaseUser: FirebaseUser)

        fun onRegistrationFailure(message: String)
    }

    interface Presenter {
        fun register(activity: Activity, email: String, password: String)
    }

    interface Interactor {
        fun performFirebaseRegistration(activity: Activity, email: String, password: String)
    }

    interface OnRegistrationListener {
        fun onSuccess(firebaseUser: FirebaseUser)

        fun onFailure(message: String)
    }
}
